aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1482,10 +1562,24 @@ impl AptosVM {
                             continue;
                         }
 
-                        let size_if_module_exists = resolver
-                            .as_executor_view()
-                            .get_module_state_value_size(&state_key)
-                            .map_err(|e| e.finish(Location::Undefined))?;
+                        let size_if_module_exists = if self.features().use_loader_v2() {
+                            if module_storage
+                                .check_module_exists(addr, name)
```

Comment:
Hm, how about we use `Result<Option<T>>` pattern here then? My goal was to avoid remapping option to linker error on Move side, but for efficiency returning an option can be better compared to repeated calls. It would also be consistent with state value metadata and bytes (added in a later PR).

Reply:
Seems like option is better, I will just stack that on the top-level PR since they next ones are quite small, but I changed types from partial errors to errors, so would prefer not to rebase that 😂

Reply:
sure, can do in follow-ups

Reply:
Done in #14402